pkg/bgpv1/manager/reconcilerv2/neighbor.go

```diff
@@ -148,12 +188,27 @@ func (r *NeighborReconciler) Reconcile(ctx context.Context, p ReconcileParams) e
 		}
 
 		if n.PeerAddress == nil {
-			r.logger.Debug("Peer does not have PeerAddress configured, skipping", types.PeerLogField, n.Name)
-			continue
+			// future auto-discovery modes can be added here to get the peer address
+			switch n.AutoDiscovery.Mode {
+			case v2.BGPDefaultGatewayMode:
+				defaultGateway, err := r.getDefaultGateway(n.AutoDiscovery.DefaultGateway)
+				if err != nil {
+					r.logger.Debug("failed to get default gateway, skipping",
+						types.PeerLogField,
+						n.Name,
+						logfields.Error,
+						err)
+					continue
+				}
+				newNeigh[i].PeerAddress = &defaultGateway
+			default:
+				r.logger.Debug("Peer does not have PeerAddress configured, skipping", types.PeerLogField, n.Name)
+				continue
+			}
```

Comment:
Nit: We can declare the common logger at the beginning of the loop. So that we don't need to repeat it in L196 and L205.

```
for i, n := range newNeigh {
    l := r.logger.With(types.PeerLogField, n.Name)
```

Reply:
i re-used the parent logger and added the above fields to child logger which will be used within the loop